modules/phase_field/tests/misc/interface_flux.i

```diff
@@ -0,0 +1,151 @@
+#
+# This test demonstrates an InterfaceKernel (InterfaceDiffusionFlux) that can
+# replace a pair of integrated DiffusionFluxBC boundary conditions.
+# Uncomment the commented out parts of teh input file to activate teh simultaneous
+# solve of the same problem with the BCs applied instead of the InterfaceKernel.
+#
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 20
+  ny = 10
+  ymax = 0.5
+[]
+
+[MeshModifiers]
+  [./box1]
+    type = SubdomainBoundingBox
+    block_id = 1
+    bottom_left = '0 0 0'
+    top_right = '0.51 1 0'
+  [../]
+  [./box2]
+    type = SubdomainBoundingBox
+    block_id = 2
+    bottom_left = '0.49 0 0'
+    top_right = '1 1 0'
+  [../]
+  [./iface_u]
+    type = SideSetsBetweenSubdomains
+    master_block = 1
+    paired_block = 2
+    new_boundary = 10
+    depends_on = 'box1 box2'
+  [../]
+  #[./iface_v]
+  #  type = SideSetsBetweenSubdomains
+  #  master_block = 2
+  #  paired_block = 1
+  #  new_boundary = 11
+  #  depends_on = 'box1 box2'
+  #[../]
+[]
+
+[Variables]
+  #[./u1]
+  #  block = 1
+  #  [./InitialCondition]
+  #    type = FunctionIC
+  #    function = 'r:=sqrt((x-0.4)^2+(y-0.5)^2);if(r<0.05,5,1)'
+  #  [../]
+  #[../]
+  #[./v1]
+  #  block = 2
+  #  [./InitialCondition]
+  #    type = FunctionIC
+  #    function = 'r:=sqrt((x-0.7)^2+(y-0.5)^2);if(r<0.05,5,1)'
+  #  [../]
+  #[../]
+  [./u2]
+    block = 1
+    [./InitialCondition]
+      type = FunctionIC
+      function = 'r:=sqrt((x-0.4)^2+(y-0.5)^2);if(r<0.05,5,1)'
+    [../]
+  [../]
+  [./v2]
+    block = 2
+    [./InitialCondition]
+      type = FunctionIC
+      function = 'r:=sqrt((x-0.7)^2+(y-0.5)^2);if(r<0.05,5,1)'
+    [../]
+  [../]
+[]
+
+[Kernels]
+  #[./u1_diff]
+  #  type = Diffusion
+  #  variable = u1
+  #  block = 1
+  #[../]
+  #[./u1_dt]
+  #  type = TimeDerivative
+  #  variable = u1
+  #  block = 1
+  #[../]
+  #[./v1_diff]
+  #  type = Diffusion
+  #  variable = v1
+  #  block = 2
+  #[../]
+  #[./v1_dt]
+  #  type = TimeDerivative
+  #  variable = v1
+  #  block = 2
+  #[../]
+
+  [./u2_diff]
+    type = Diffusion
+    variable = u2
+    block = 1
+  [../]
+  [./u2_dt]
+    type = TimeDerivative
+    variable = u2
+    block = 1
+  [../]
+  [./v2_diff]
+    type = Diffusion
+    variable = v2
+    block = 2
+  [../]
+  [./v2_dt]
+    type = TimeDerivative
+    variable = v2
+    block = 2
+  [../]
+[]
+
+[InterfaceKernels]
+  [./iface]
+    type = InterfaceDiffusionFlux
+    boundary = 10
+    variable = u2
+    neighbor_var = v2
+  [../]
+[]
+
+#[BCs]
+#  [./u_boundary_term]
+#    type = DiffusionFluxBC
+#    variable = u1
+#    boundary = 10
+#  [../]
+#  [./v_boundary_term]
+#    type = DiffusionFluxBC
+#    variable = v1
+#    boundary = 11
+#  [../]
+#[]
+
```

Comment:
See top comment in that file. I'd like to retain the objects that reproduce the same result. But I didn't want to bloat the gold file :-/